### PR TITLE
[FIX]Refactor Leaderboard

### PR DESCRIPTION
**Pull request by samad-yar-khan**
-  [x] I have read the Contributing Guide - https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat doc
  -  [x] I have signed the CLA - https://cla-assistant.io/RocketChat/Rocket.Chat
  - [x] Lint and unit tests pass locally with my changes
  - [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
  -  [ ] I have added necessary documentation (if applicable)
  - [x] Any dependent changes have been merged and published in downstream modules

## Issues

#75 

## Proposed changes (including videos or screenshots)

- Made separate `leaderboard.js `and `leaderboardcompact.js` components to improve re-usability of the component by community builders.

https://user-images.githubusercontent.com/70485812/152694865-03dcb908-d460-4680-b564-a8e7be653327.mp4

### CODE REVIEW

File: app/pages/gsoc/gsoc2022/[id].js [hunk 1]
```diff
@@ -1,11 +1,15 @@
 import Head from "next/head";
+import Image from "next/image";
 import { fetchAPI } from "../../../lib/api";
 import { contributorList } from "../../../lib/leaderboard";
-import { Container, Col } from "react-bootstrap";
+import { Container, Col, Table } from "react-bootstrap";
 import styles from "../../../styles/Leaderboard.module.css";
```
Comment (Sing-Li) [inline app/pages/gsoc/gsoc2022/[id].js:6]: This is part of the component, it should not need to be imported here.

File: app/pages/gsoc/gsoc2022/[id].js [hunk 2]
```diff
@@ -1,11 +1,15 @@
 import Head from "next/head";
+import Image from "next/image";
 import { fetchAPI } from "../../../lib/api";
```
Comment (Sing-Li) [inline app/pages/gsoc/gsoc2022/[id].js:3]: Fetch of data should be part of the component, should not need to be imported here.

File: app/pages/gsoc/gsoc2022/[id].js (hunk 1 above)
Comment (Sing-Li) [inline app/pages/gsoc/gsoc2022/[id].js:6]: This is a "demonstration page" for the components.

The import should import the components ONLY and then laid them out using simple JSX / bootstrap.

I suggest you include 1 instance of the full size   `leaderboard`   and 2 instances of the `leaderboardcompact` on this page.

**Review by Sing-Li — CHANGES_REQUESTED** [review]

File: app/pages/gsoc/gsoc2022/[id].js (hunk 2 above)
Comment (samad-yar-khan) [reply app/pages/gsoc/gsoc2022/[id].js:3]: Sir, this fetchAPI is needed to fetch the top nav-bar and to make the dynamic routes possible. It is for the strapi APIs.

File: app/pages/gsoc/gsoc2022/[id].js (hunk 1 above)
Comment (samad-yar-khan) [reply app/pages/gsoc/gsoc2022/[id].js]: I have hidden these inside the leaderboard component now. No CSS leaks on the page anymore :)

**Pull request merged**